### PR TITLE
Squash ug bugs, update ug, update help window content

### LINKED ISSUE #145: [PE-D][Tester C] Correctness of add command in UG's "Adding a client" section
In the second example given in the UG the phone number "p/1234567" is invalid . Upon entering that example, an error message appears stating that the phone number should be a SG number with exactly 8 digits. I am not sure whether if this 2nd example is intended to be a bad example for user demonstration or is this a typo. I think it would be better to expected result of running the command to avoid confustion. <br> `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567
 b/11 09 2001 t/criminal`

 

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: IanCheah/ped#1

### LINKED ISSUE #151: [PE-D][Tester A] Bad line wrapping in Saving and Editing Data section of help window
-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: 1st2GetThisName/ped#2

### LINKED ISSUE #152: [PE-D][Tester E] Usage message for add command is inconsistent with that in the UG
The usage message for add command should have a square bracket surrounding a/ADDRESS to be in line with the UG, so that users are aware that the address field is optional

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: cshao02/ped#4

### LINKED ISSUE #155: [PE-D][Tester C] Spelling error in "Editing a client" section
There is a small grammar mistake here. It should be `Need to update a client's details?`, because the letter c is not a vowel.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: IanCheah/ped#4

### LINKED ISSUE #164: [PE-D][Tester D] UG: Command summary and command details is not aligned
Steps to reproduce: 

1. The edit command in command summary states
    
    edit INDEX [n/NAME] **[p/PHONE_NUMBER]** [e/EMAIL] [a/ADDRESS] [t/TAG]…
    
    while the edit command details gives 
    
    edit INDEX [n/NAME] **[p/PHONE]** [e/EMAIL] [a/ADDRESS] [b/BIRTHDAY] [t/TAG]…
    

Problematic behavior: user may confuse on the input of phone number or other info regarding phone like phone model etc

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: SamsonChew/ped#1

### LINKED ISSUE #165: [PE-D][Tester D] Valid name input but fail to execute add command
Steps to reproduce:

1. input add command:
    
    add n/John A/P Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 b/11 09 2001 t/friends t/owesMoney
    

Problematic behavior: John A/P Doe as a legal name is not accepted in the program

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: SamsonChew/ped#2

### LINKED ISSUE #172: [PE-D][Tester D] The example command in User Guide is not working for find command
Steps to reproduce:

1. reference to the paid command detail example command asks us to use 
    
    “find Betsy followed by paid 1 f/6 marks the 1st client in the results of the find command as
    
    paid.”
    
2. use command of: find John gives me invalid command format 

Problematic behavior: Incorrect command example that leads to unexpected behavior
Suggestion: update the UG for correction for the format for find command

The keyword

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: SamsonChew/ped#7

### LINKED ISSUE #180: [PE-D][Tester D] DG - incomplete glossary
Steps to reproduce:

1. compare the glossary in user guide & developer guide

Suggestion behavior: Try to align with both glossary in UG & DG so that more terms can be included from what I observe is that some terms in UG is not in DG eg. CLI (in UG but not DG) while GUI (in DG but not in UG)

Developer guide glossary

User guide glossary

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: SamsonChew/ped#12

### LINKED ISSUE #183: [PE-D][Tester D] Missing upload command in both command details and command summary
Steps to reproduce:

1. Check User guide for the section “Upload a client’s profile picture [WINDOWS ONLY]”
2. Check the glossary section (missing of upload command detail)

Problematic behavior: That may confuse the reader for the correct command to be used

Suggested behavior: align with the previous commands maybe in “**Upload a client’s profile picture [WINDOWS ONLY]: upload”**

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: SamsonChew/ped#15

### LINKED ISSUE #192: [PE-D][Tester E] UG edit command missing explanation on usage of a/ to remove a person's address
Perhaps for the UG edit command section, a line can be added to explain that typing `a/` without specifying an address can also likewise remove the person's address, as address is also an optional field like tags.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: cshao02/ped#9

### LINKED ISSUE #205: [PE-D][Tester C] Tracking of payment status
It is mentioned that in the UG under the  `Marking a client as paid` section that `Need to track which clients have paid their policy premiums for the current period?` So for example if I have person A who have a payment cycle of 12 months, that starts from January, another person B who have a payment cycle of 12 months that starts on Feburary. Suppose I were to indicate both of their paid status as paid, how would I know when to mark it back as unpaid? This will be difficult to rememeber if there are many clients. I have several questions about how this works.

- Is there an assumption that the starting month of every payment cycle is the same?
- Will there be a proposed feature that handles this case. If yes, I think you guys can include it to prevent something similar from being raised in actualy PE.

Currently, it seems to me that marking and unmarking does not really help in solving the problem of `tracking`.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: IanCheah/ped#11

### LINKED ISSUE #216: [PE-D][Tester B] No information given in UG about template message in client details displayed in right panel
### Use of the template message buttons is unknown as no information is provided about it in the UG.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: Anvita2110/ped#15

### LINKED ISSUE #218: [PE-D][Tester B] Title for UG should have been "F.A.R.T"
-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Anvita2110/ped#17

### LINKED ISSUE #221: [PE-D][Tester B] Order of optional and required parameters in UG
### Since there is no order of parameters in commands, b/birth date should have been shown before the optional parameters as it might go unnoticed for a novice user or first time user as it happened with me too (and would not be aware of it until he/she gets invalid command error)
---

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Anvita2110/ped#19

### LINKED ISSUE #223: [PE-D][Tester B] Minimum Screen Resolution does not show Command entering and result panel
### Suggestion: Minimum resolution window should show all the panels currently in it
---

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: Anvita2110/ped#21

### PR DESCRIPTION
**Pull request by Shantaa**
Closes #205: Add clarifications under Paid and Unpaid sections of UG
Closes #180: Add definition under Glossary section of UG
Closes #218: Update UG title to F.A.R.T.
Closes #216: Add template message explanation under Viewing a client's details section of UG
Closes #223: Add under Known Issues section of UG
Closes #152: Add optional indicator for Address field
Closes #192: Add into Edit explanation
Closes #183: Add upload command into Command Summary
Closes #172: Edit find examples to reflect new find format
Closes #221: Update order of fields to put required ones first
Closes #151: Wrap lines better
Closes #145: Update example to be correct
Closes #155: Update grammar
Closes #164: Update field from PHONE to PHONE_NUMBER

Add a Validation Rules section to the UG, explaining valid and invalid inputs of each field
Closes #165

Reorder sections of the UG for better clarify and flow of document
Update images to use latest versions
Add spaces in front of headers and used generic numbering for ordered lists, as recommended by Markdown Syntax Guide

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -3,56 +3,66 @@ layout: page
...
 
 1. Ensure you have Java `17` or above installed in your Computer.
    1. You can check which version of Java you are currently running by entering `java -version` into a command terminal.
-   2. Should you require help, [here](https://www.oracle.com/java/technologies/downloads/#java17) is the download link to Java '17'
-   3. After accessing the website, please choose the right download link for your operating system (Linux, macOS or Windows)
+   1. Should you require help, [here](https://www.oracle.com/java/technologies/downloads/#java17) is the download link to Java 17
+   1. After accessing the website, please choose the right download link for your operating system (Linux, macOS or Windows)
 
-2. Download the latest `.jar` file from [here](https://github.com/AY2425S1-CS2103T-F14b-4/tp/releases/tag/v1.4).
+1. Download the latest `.jar` file from [here](https://github.com/AY2425S1-CS2103T-F14b-4/tp/releases/tag/v1.4).
```
Comment (meatyturtle) [inline docs/UserGuide.md:32]: the latest .jar file is v1.6, unfortunately will have to change this right before submitting final version

File: docs/UserGuide.md [hunk 2]
```diff
@@ -76,75 +86,153 @@ Financial Assurance Revolutionary Telemarketer (F.A.R.T) is a **desktop app for
...
+* `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/8234567 b/11 09 2001 t/criminal`
 
 **Note:**
-1. As of version 1.5, only valid email address (in the form name@domain.com) and **Singapore** phone numbers are accepted.
-Birthdays should be added in DD MM YYYY format.
-2. You may add duplicate contacts (i.e., contacts with the same name, case-insensitive) but the F.A.R.T book will prompt you
-to confirm that you are adding a duplicate contact.
+1. As mentioned in the Validation Rules, only valid email addresses (in the form `local-part@domain`) and **Singapore** phone numbers are accepted.
+1. Birthdays should be added in DD MM YYYY format.
+1. You may add duplicate contacts (i.e., contacts with the same name, case-insensitive) but the F.A.R.T book will prompt you to confirm that you are adding a duplicate contact. If you select `No` in the prompt window, the operation will be aborted and the client will not be added. Else, select `Yes` to add the new client.
```
Comment (meatyturtle) [inline docs/UserGuide.md:186]: same index?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -156,42 +244,55 @@ Format: `delete INDEX`
...
 Examples:
 * `list` followed by `delete 2` deletes the 2nd client in the F.A.R.T book.
-* `find Betsy` followed by `delete 1` deletes the 1st client in the results of the `find` command.
+* `find n/Betsy` followed by `delete 1` deletes the 1st client in the results of the `find` command.
 
 **Note:**
-1. As a `delete` command cannot be undone, the F.A.R.T Book will first prompt you to confirm if the
- contact that you want to delete is correct. If you select `No`, the operation will be aborted and the client will not be deleted.
+1. A `delete` command cannot be undone, and deleted contacts cannot be retrieved.
+1. When a `delete` command is inputted, the F.A.R.T Book will first prompt you to confirm if the contact that you want to delete is correct. If you select `No`, the operation will be aborted and the client will not be deleted. Else, select `Yes` to delete the contact.
```
Comment (meatyturtle) [inline docs/UserGuide.md:251]: should change contact to client instead in this case

File: docs/UserGuide.md [hunk 4]
```diff
@@ -156,42 +244,55 @@ Format: `delete INDEX`
...
 Format: `clear`
 
 **Note:**
-1. A `clear` command cannot be undone, the F.A.R.T Book will first prompt you to confirm that the
-   you wish to clear the book. If you select `No`, the clear action will be aborted.
-2. If you do change your mind after clearing the F.A.R.T book, you may retrieve the sample data again
-by deleting the `addressbook.json` file in the `data` folder, then launching F.A.R.T again.
+1. A `clear` command cannot be undone, and cleared contacts cannot be retrieved.
+1. When a `clear` command is inputted, the F.A.R.T Book will first prompt you to confirm that you wish to clear the book. If you select `No`, the operation will be aborted and the book will not be cleared. Else, select `Yes` to clear the book.
+1. If you do change your mind after clearing the F.A.R.T book, you can retrieve only the sample data by deleting the `addressbook.json` file in the `data` folder, then launching F.A.R.T again.
```
Comment (meatyturtle) [inline docs/UserGuide.md:265]: do the list numberings show up as 1, 2, 3...  after parsing?

File: docs/UserGuide.md (hunk 2 above)
Comment (Shantaa) [reply docs/UserGuide.md:186]: Yes, will be numbered accordingly when converting.

File: docs/UserGuide.md (hunk 3 above)
Comment (Shantaa) [reply docs/UserGuide.md:251]: Noted on this.

**Pull request merged**